<launch>
    <!-- odometer delta input topic -->
    <arg name="odo_topic" default="/from_mav/odometer_delta"/>
    <!-- IMU delta input topic -->
    <arg name="imu_topic" default="/from_mav/imu"/>

    <!-- rate at which the odometry calculation will run (limited by the input topics rates) -->
    <arg name="rate" default="80"/>
    <!-- size of subscriber and sync policy queues -->
    <arg name="queue_size" default="5"/>

    <!-- tf parent frame of published tf -->
    <arg name="tf_parent" default="odometry"/>
    <!-- tf child frame of published tf -->
    <arg name="tf_child" default="rear_axis_middle_ground"/>

    <!-- debug rviz marker -->
    <arg name="debug_rviz" default="true" />


    <node name="imu_ros_odometry" pkg="drive_ros_imu_odo_odometry" type="drive_ros_imu_odo_odometry_node" output="screen">
        <remap from="~/odo_in" to="$(arg odo_topic)"/>
        <remap from="~/imu_in" to="$(arg imu_topic)"/>
        <param name="rate" type="int" value="$(arg rate)" />
        <param name="queue_size" type="int" value="$(arg queue_size)" />
        <param name="tf_parent" type="str" value="$(arg tf_parent)" />
        <param name="tf_child" type="str" value="$(arg tf_child)" />
        <param name="debug_rviz" type="bool" value="$(arg debug_rviz)" />
        <rosparam command="load" file="$(find drive_ros_imu_odo_odometry)/config/kalman_cov.yaml"/>
        <rosparam command="load" file="$(find drive_ros_imu_odo_odometry)/config/sensor_cov.yaml"/>
    </node>
</launch>
